### ISSUE TITLE
UPM Package without ExternalDependencyManager

### ISSUE THREAD

**Issue by snorrsi** [amplitude/unity-plugin#111]
## Summary

I noticed that you added ExternalDependencyManager into your tagged versions.
I suppose that is because Google still hasn't bothered to create a Github package download for that.
However, I suppose most of us that are using your package are in one way or another including ExternalDependencyManager in our projects. Which means that when including your package through github url (or other means) will lead to conflict with guids.

To solve this, Can you please create a separate tags for your plugin that have ExternalDependencyManager not added ?

## Motivations

Creating separate tags that does not include ExternalDependencyManager solves guid conflicts for those already including it.

**Comment by snorrsi** [comment]
Hey @dnshi as you added this maybe you comment on this issue of mine :)

**Comment by JonLandmark** [comment]
I would also like to request that ExternalDependencyManager not be included in this package. I already have the OpenUPM ExternalDependencyManager as a package in my project so the latest Amplitude SDK will conflict if I update it. Also, conflicts will occur if other SDKs have their own version of the ExternalDependencyManager.

**Comment by magicfinger96** [comment]
sorry ive created the same issue earlier. I would like you to remove the EDM4U from your package too. When are you going to do this? It's been almost a month.

**Comment by qingzhuozhen** [comment]
In v2.6.0, we released amplitude-unity.unitypackage without ExternalDependencyManager; and amplitude-unity-all.unitypackage with ExternalDependencyManager in case any one need it. Please let us know if this helps or any other suggestions. Thanks for raising the concern.

**Comment by snorrsi** [comment]
> In v2.6.0, we released amplitude-unity.unitypackage without ExternalDependencyManager; and amplitude-unity-all.unitypackage with ExternalDependencyManager in case any one need it. Please let us know if this helps or any other suggestions. Thanks for raising the concern.

Doesn´t help for us that use UPM.
Both cases are unresolved there as both ExternalDepency folder and Plugins/Android folder are still there.
It would help if you could create UPM special tag where these folders and meta files are removed and perhaps even only including the conent under /Assets folder to skip having to specify path in UPM

**Comment by qingzhuozhen** [comment]
@snorrsi Thanks for the feedback. We will work on supporting this.

**Comment by snorrsi** [comment]
@qingzhuozhen thanks .. I also noticed that there is unneeded Plugins/AmplitudeSDKPlugins.asmdef.meta at https://github.com/amplitude/unity-plugin/blob/main/Assets/Plugins/AmplitudeSDKPlugins.asmdef.meta

It can safely be deleted  as Plugins/AmplitudeSDKPlugins.asmdef does not exist (and is not needed as there is no .cs files in the folder or any of it's subfolder) and Unity dislikes meta files without corresponding file

**Comment by qingzhuozhen** [comment]
Hi @snorrsi, thanks for raising this up. I cut a tag `v2.6.0-no-edm` without ExternalDependencyManager and the files you mentioned. Please let us know if that helps.

**Comment by snorrsi** [comment]
> Could you give a UPM link for v2.6.0-no-edm Like https://github.com/amplitude/unity-plugin?path=/Assets but for v2.6.0-no-edm
> 
> Or through openupm https://openupm.com/packages/com.amplitude.unityplugin/?subPage=pipelines

@shamruk that link would be

https://github.com/amplitude/unity-plugin.git?path=/Assets#v2.6.0-no-edm

or for the other branch without dependencies

https://github.com/amplitude/unity-plugin.git?path=/Assets#v2.6.0-no-edm-and-dep

have tested neither though as still using my own version

**Comment by shamruk** [comment]
[snorrsi](https://github.com/snorrsi) 
https://github.com/amplitude/unity-plugin.git?path=/Assets#v2.6.0-no-edm
opens main branch with edm

i'll go with creating own fork for now (just merged tag into main there)
https://github.com/shamruk/unity-plugin.git?path=/Assets
PS also restored asmdef files

**Comment by snorrsi** [comment]
@shamruk these link only work in Unity Package Manager. The links don't work in browser, at least the branch name is not changed and remains the default one of the repository.

See https://docs.unity3d.com/Manual/upm-git.html#revision

**Comment by qingzhuozhen** [comment]
Thanks @snorrsi for sharing the instructions for tag or branch support. I will update it in our developer doc as well. Also I add the unity package for [v2.6.0-no-edm](https://github.com/amplitude/unity-plugin/releases/tag/v2.6.0-no-edm) and [v2.6.0-no-edm-and-dep](https://github.com/amplitude/unity-plugin/releases/tag/v2.6.0-no-edm-and-dep). Hope these can help.

**Comment by shamruk** [comment]
@snorrsi Oh, didn't know that, thanks! 
Anyway, I'll wait asmdef (separate task) fixed before switching to the link.

@qingzhuozhen If possible also look into openupm, pipeline looks broken because the version is the same.
https://openupm.com/packages/com.amplitude.unityplugin/?subPage=pipelines
PS openupm looks like the most professional and clean way to add extensions

**Comment by lordmortis** [comment]
Is there a version of 2.8 without EDM?

**Comment by lordmortis** [comment]
Yes! See [here](https://github.com/amplitude/unity-plugin/issues/116#issuecomment-2203812077) for info!
